test/api/v3/integration/groups/POST-groups_groupId_join.test.js

```diff
@@ -203,6 +203,13 @@ describe('POST /group/:groupId/join', () => {
         await expect(invitedUser.get('/user')).to.eventually.have.nested.property('party._id', party._id);
       });
 
+      it('accepting a redundant party invite will let the user stay in the party', async () => {
+        await invitedUser.post(`/groups/${party._id}/join`);
+        await invitedUser.post(`/groups/${party._id}/join`);
```

Comment:
@paglias 
I tried to execute the repro steps of this issue i.e., sending the invite, manually adding the user to the party, and then executing the POST call for joining the party. This should be sufficient for testing the new changes.

However, I am not able to pre-add the member to the party. If I try to add using a POST request, then I get an error for either of the 2 reasons - 

1. Invitation cannot be sent as member is already in the party. 
2. A previous invite already exists.

I am not sure how to go ahead with this test. Can you help me out? 
Is there a way by which I can add the specific UUID to the party members list?

Reply:
Yeah, so the API in normal circumstances will block you from inviting someone already part of your group. I think the only way to test it would be to directly editing the user document using `await invitedUser.update(... changes go here ...)` you can see in the existing tests how it's used for example here https://github.com/HabitRPG/habitica/blob/f8bab04a0e66a819f6d7ca5b210d27d991446bf3/test/api/v3/integration/user/POST-user_purchase.test.js#L38

Reply:
@paglias Thanks for the help! :D 

Please have a look at the PR and let me know if we need any other changes.